### PR TITLE
Add .gitlab-ci.yml 

### PR DESCRIPTION
**Pull request by jgmize**
- mozmeao/infra#1152
- Run git push conditionally within update_docker.sh
- https://gitlab.com/mozmeao/release-notes/-/jobs/381130486
- after merge:
  - [x] disable Jenkins job
  - [x] kubectl apply -f k8s/cron.yaml

### CODE REVIEW

**Comment by pmac** [comment]
My only hesitation is that, if I'm reading this right, the fastest schedule that can be run on gitlab.com is once per hour, due to [this setting](https://docs.gitlab.com/ee/user/gitlab_com/index.html#cron-jobs). We currently run the update every 5 or 15 min I believe. This will mostly be fine, but it means we may have to manually kick off an update or two on release days. Moving nucleus to directly commit to git will also fix this, so I hope to accelerate that work, but for now this could be an issue.

File: update_docker.sh [hunk 1]
```diff
@@ -17,15 +17,11 @@ fi
...
-# UPDATE_FILE is an indicator we can use in Jenkins
-# to run other jobs if there was, in fact, an update
-rm -f "$UPDATE_FILE"
-
 if [[ "$1" == "commit" ]]; then
     if git status --porcelain | grep -E "\.json$"; then
         git add ./releases/
         git commit -m "Update release data"
-        git rev-parse HEAD > "$UPDATE_FILE"
+        git push
```
Comment (pmac) [inline update_docker.sh]: to which repo will this push? I think with this setup that would be Gitlab? We'll probably need to do something like `git push git@github.com/mozilla/release-notes.git`

**Review by pmac — CHANGES_REQUESTED** [review]
See my comment. I think we need to do something to ensure it's pushing to the right git repo.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+update:
+  only:
+    - master
```
Comment (pmac) [inline .gitlab-ci.yml]: I don't think we want this to run on commits to master since this job commits to master.

**Comment by pmac** [comment]
We could potentially quickly add triggering the Gitlab pipeline via a nucleus post-save signal, or even cron running somewhere, and use `only: triggers` in the config:

https://docs.gitlab.com/ee/ci/yaml/README.html#onlyexcept-basic

File: .gitlab-ci.yml (hunk 2 above)
Comment (jgmize) [reply .gitlab-ci.yml]: good catch! removing now.

File: k8s/cron.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          containers:
+          - name: curl
+            image: curlimages/curl
+            args:
+            - -X
+            - POST
+            - -F
+            - token=$(TOKEN)
+            - -F
+            - ref=gitlab
```
Comment (jgmize) [inline k8s/cron.yaml]: - [ ] change this to `ref=master` after this is merged.

**Comment by jgmize** [comment]
Thanks for the excellent feedback and suggestions, @pmac! I've applied your suggested changes and `k8s/cron.yaml` from 7f053a4 in iowa-a and it is triggering the [pipelines](https://gitlab.com/mozmeao/release-notes/pipelines) in the gitlab branch. Once this is merged I'll apply the change in b02b5db so that it triggers the pipelines in the master branch.

**Pull request merged**